### PR TITLE
Animation Assemblers!

### PR DESCRIPTION
**Pull request by avaidyam**
Pre-built animations with only one line of code. Why? More like why not? 

You can use them if you want, and if you chose to, they're only one line of code! And if you're not animating `TUIViews`, you can use the `TUIAnimationAssembler` to get a custom `CAAnimation` or `CAAnimationGroup` to apply elsewhere. No cost to you if you chose not to use them, too.

P.S. Comes with full documentation this time :P

### CODE REVIEW

**Comment by jspahrsummers** [comment]
I really like this idea, but this seems like a better extension to TwUI than something that should be within the framework itself. The framework should include the basics for getting a modern and CALayer-based UI, and then there can of course be other frameworks that build on TwUI (just like there are for AppKit and UIKit).

TL;DR: I think this is out of scope.

**Comment by avaidyam** [comment]
Alright, makes sense that way.
